cli/azd/internal/appdetect/appdetect_test.go

```diff
@@ -193,7 +193,8 @@ func TestDetectDocker(t *testing.T) {
 		Path:          filepath.Join(dir, "dotnet"),
 		DetectionRule: "Inferred by presence of: dotnettestapp.csproj, Program.cs",
 		Docker: &Docker{
-			Path: filepath.Join(dir, "dotnet", "Dockerfile"),
+			Path:  filepath.Join(dir, "dotnet", "Dockerfile"),
+			Ports: nil,
```

Comment:
Let's also update this test case to write different port handling conditions `Dockerfile`:

I recommend creating a new subtest within `TestDetectDocker` with:

```golang
t.Run("exposed ports", func(t *testing.T) {
})
```

I would think about using the same type of table-driven tests you had written for `docker_test.go`. Each test case specifies the `exposeLine` to write to the Dockerfile to set up the test case:

- Single port
- Multiple ports on single line
- Multiple ports on multiple lines
- No ports

Reply:
Added `TestParsePortsInFile` in `docker_test.go`